conans/client/remote_manager.py

```diff
@@ -61,12 +61,17 @@ def upload_recipe(self, ref, remote, retry, retry_wait, policy, remote_manifest)
         if policy == UPLOAD_POLICY_SKIP:
             return ref
 
-        ret, rev_time = self._call_remote(remote, "upload_recipe", ref,
-                                          the_files, retry, retry_wait, policy, remote_manifest)
+        ret = self._call_remote(remote, "upload_recipe", ref, the_files, retry, retry_wait,
+                                policy, remote_manifest)
 
         # Update package revision with the rev_time (Created locally but with rev_time None)
         with self._cache.package_layout(ref).update_metadata() as metadata:
-            metadata.recipe.time = rev_time
+            if metadata.recipe.time is None:
+                _, _, rev_time = self._call_remote(remote, "get_recipe_snapshot", ref)
+                # Only if we hadn't time locally update it, otherwise we are uploading
+                # the recipe to a different server, but in the upload never the registry
+                # is updated, so the time should be kept to the first upload
+                metadata.recipe.time = rev_time
```

Comment:
I think last time we talked, we agreed that uploading didn't update the time, only installing from the server.

Reply:
You are right. But the new PR will change all this. This PR is older than that conversation. Keep that thought until the new PR.